File: demo/src/main/java/com/fengshuisystem/demo/entity/Account.java
package com.fengshuisystem.demo.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fengshuisystem.demo.entity.enums.Gender;
import com.fengshuisystem.demo.entity.enums.Status;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;
import org.hibernate.annotations.Nationalized;
import org.springframework.security.core.context.SecurityContextHolder;

import java.time.Instant;
import java.time.LocalDate;
import java.util.LinkedHashSet;
import java.util.Set;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "account_id", nullable = false)
    private Integer id;

    @Size(max = 50)
    @Column(name = "username", length = 50)
    private String userName;

    @Size(max = 255)
    @Column(name = "password")
    private String password;

    @Size(max = 255)
    @Column(name = "fullname")
    private String fullName;

    @Size(max = 255)
    @Nationalized
    @Column(name = "email")
    private String email;

    @Size(max = 255)
    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "gender")
    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Lob
    @Column(name = "avatar")
    private String avatar;

    @Column(name = "dob")
    private LocalDate dob;

    @Column(name = "code")
    private String code;

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private Status status = Status.INACTIVE;

    @Column(name = "created_date")
    private Instant createdDate;

    @Size(max = 50)
    @Column(name = "created_by",  length = 50)
    private String createdBy;

    @Column(name = "updated_date")
    private Instant updatedDate;

    @Size(max = 50)
    @Column(name = "updated_by", length = 50)
    private String updatedBy;

    private String fcmToken;

    @ManyToMany(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinTable(name = "account_role",
            joinColumns = @JoinColumn(name = "account_id"),
            inverseJoinColumns = @JoinColumn(name = "role_id"))
    private Set<Role> roles = new LinkedHashSet<>();

    @OneToMany(mappedBy = "account", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<Bill> bills = new LinkedHashSet<>();

    @OneToMany(mappedBy = "account", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<ConsultationRequest> consultationRequests = new LinkedHashSet<>();

    @OneToMany(mappedBy = "account", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<ConsultationResult> consultationResults = new LinkedHashSet<>();

    @OneToMany(mappedBy = "account", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<Post> posts = new LinkedHashSet<>();
}

File: src/models/Account.ts
import { Gender } from './enums/Gender';
import { Status } from './enums/Status';
import { Role } from './Role';
import { Bill } from './Bill';
import { ConsultationRequest } from './ConsultationRequest';
import { ConsultationResult } from './ConsultationResult';
import { Post } from './Post';

export class Account {
    accountId: number;
    userName: string;
    password: string;
    fullName: string;
    email: string;
    phoneNumber: string;
    gender: Gender | null;
    avatar: string | null;
    dob: Date | null;
    code: string | null;
    status: Status;
    createdDate: Date;
    createdBy: string;
    updatedDate: Date;
    updatedBy: string;
    roles: Set<Role>;
    bills: Set<Bill>;
    consultationRequests: Set<ConsultationRequest>;
    consultationResults: Set<ConsultationResult>;
    posts: Set<Post>;

    constructor() {
        this.accountId = 0;
        this.userName = '';
        this.password = '';
        this.fullName = '';
        this.email = '';
        this.phoneNumber = '';
        this.gender = null;
        this.avatar = null;
        this.dob = null;
        this.code = null;
        this.status = Status.INACTIVE;
        this.createdDate = new Date();
        this.createdBy = '';
        this.updatedDate = new Date();
        this.updatedBy = '';
        this.roles = new Set<Role>();
        this.bills = new Set<Bill>();
        this.consultationRequests = new Set<ConsultationRequest>();
        this.consultationResults = new Set<ConsultationResult>();
        this.posts = new Set<Post>();
    }
}
